CONTRIBUTING.md

```diff
@@ -0,0 +1,12 @@
+# Guide for Contributors
+
+## Pushing Changes
+
+1. Read the [Versioning Scheme of this Document](https://github.com/chipsalliance/firrtl-spec/blob/main/spec.md#versioning-scheme-of-this-document)
+2. All commit messages have a [nfc|patch|minor|major] leading tag. This can be added after-the-fact to the PR title as this will be re-used for a squash-and-merge flow.
```

Comment:
It may be good to clarify that the tag should be `[nfc]` or `[patch]` etc. Or, that was what I was intending.

This PR would then be `[nfc] Create Contributing.md`.